### PR TITLE
142->backend-master

### PR DESCRIPTION
**Pull request by SelmanB**
With all features implemented, we as the backend team have decided to merge the latest branches into backend-master.

### CODE REVIEW

**Review by EAltunoglu — APPROVED** [review]
Runned on local without errors.

**Review by myasins — APPROVED** [review]
Tests are OK. Convention is kept.

File: source/backend/api/path/views.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def post(self, request):
         return Response({'pathID': str(id)}, status=status.HTTP_200_OK)
 
 class FinishMilestone(APIView):
-    permission_classes = [IsAuthenticated]
+    #permission_classes = [IsAuthenticated]
```
Comment (SelmanB) [inline source/backend/api/path/views.py:76]: This must be uncommented to avoid unauthorized uses.

Comment (SelmanB) [reply source/backend/api/path/views.py]: The problem is resolved

**Pull request merged**